src/async_client.rs

```diff
@@ -398,48 +420,48 @@ impl AsyncClient {
 
 impl Drop for AsyncClient {
     fn drop(&mut self) {
-        // The background task handle may already have been consumed in `disconnect()`. If so, there
-        // is nothing we need to do here, the task is already being run to completion (even when the
-        // method call might have been canceled).
+        // Notify background task to cancel itself, even when [`UA_Client_run_iterate()`] would want
+        // to keep on running. This is okay: we are not issuing asynchronous requests anymore anyway
+        // (the only other call will be `UA_Client_delete()` when dropping the inner client).
+        self.background_canceled.store(true, Ordering::Relaxed);
+
+        // We need to wait for the task to finish and must do so blockingly. `UA_Client_delete()` is
+        // not safe run concurrently while `UA_Client_run_iterate()` is blocking.
         if let Some(background_handle) = self.background_handle.take() {
-            // Abort background task (at the interval await point).
-            background_handle.abort();
+            log::info!("Waiting for background task to finish");
```

Comment:
These 2 lines could be extracted into a function and shared with the explicit invocation of disconnect. The log message would then appear in both cases.

Writing another log message after `join()` returned could be helpful to analyze unexpectedly long blocking durations. There are already logs within the background tasks upon termination, but logging again after joining would terminate the span.

Or use `tracing` for this purpose which might provide more insights and avoids cluttering the code with log statements?